atom/browser/api/atom_api_app.h

```diff
@@ -228,6 +232,10 @@ class App : public AtomBrowserClient::Delegate,
 
   base::FilePath app_path_;
 
+  util::Promise* gpuinfo_promise_;
+  bool has_complete_gpu_info_;
+  bool is_fetching_gpuinfo_;
```

Comment:
A truthy check on `gpuinfo_promise_` could replace this variable right?

Reply:
Not exactly. Also actually I'm still working on the changes.

Reply:
Yes, if we return the same promise, then we don't need the variable.

Reply:
please initialize the values here, we are using C++11 class member variable initialization:
https://github.com/electron/electron/pull/13030